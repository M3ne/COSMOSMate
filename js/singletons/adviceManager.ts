/* /////////////////////////////////////////////////////////////////////////////

(c) 2016-2020, Theodore Kruczek

advice-module.js manages all recommended actions to the user in a semi-tutorial
manner. It works closely with ui.js
http://keeptrack.space

All rights reserved. No part of this web site may be reproduced, published,
distributed, displayed, performed, copied or stored for public or private
use, without written permission of the author.

No part of this code may be modified or changed or exploited in any way used
for derivative works, or offered for sale, or used to construct any kind of database
or mirrored at any other location without the express written permission of the author.

///////////////////////////////////////////////////////////////////////////// */

import { getEl } from '../lib/get-el';
import { keepTrackApi } from '../keepTrackApi';

export class AdviceManager {
  private helpHeaderDOM: HTMLElement;
  private helpOuterDOM: HTMLElement;
  private helpTextDOM: HTMLElement;
  private isAdviceEnabled = true;
  private isAdviceOpen = false;
  private tutIconDOM: HTMLElement;

  public clearAdvice(): void {
    this.helpHeaderDOM.classList.remove('help-header-sel');
    this.helpHeaderDOM.onclick = null;
  }

  public init() {
    this.helpOuterDOM = getEl('help-outer-container');
    this.helpHeaderDOM = getEl('help-header');
    this.helpTextDOM = getEl('help-text');
    this.tutIconDOM = getEl('tutorial-icon');

    this.tutIconDOM.addEventListener('click', function () {
      keepTrackApi.methods.onHelpMenuClick();
    });

    window.onkeydown = (e: KeyboardEvent) => {
      // If Shift + F1
      if (e.shiftKey && e.code === 'F1') {
        if (this.isAdviceOpen) {
          this.isAdviceOpen = false;
          this.helpOuterDOM.style.display = 'none';
        } else {
          keepTrackApi.methods.onHelpMenuClick();
        }
      }
    };
  }

  public isEnabled(): boolean {
    return this.isAdviceEnabled;
  }

  public off() {
    try {
      localStorage.setItem('isAdviceEnabled', 'false');
    } catch {
      // Do Nothing
    }
    this.isAdviceEnabled = false;
    this.isAdviceOpen = false;
    this.helpOuterDOM.style.display = 'none';
    this.tutIconDOM.classList.remove('bmenu-item-selected');
  }

  public on() {
    try {
      localStorage.setItem('isAdviceEnabled', 'true');
    } catch {
      // Do Nothing
    }
    this.isAdviceEnabled = true;
    this.tutIconDOM.classList.add('bmenu-item-selected');
  }

  public showAdvice(header: string, text: string): void {
    if (!this.isAdviceEnabled) return;

    this.isAdviceOpen = true;

    this.clearAdvice();

    this.helpOuterDOM.style.display = 'block';
    this.helpHeaderDOM.innerHTML = header;
    this.helpTextDOM.innerHTML = text;
    this.helpOuterDOM.addEventListener('click', () => {
      this.isAdviceOpen = false;
      this.helpOuterDOM.style.display = 'none';
    });
  }
}

export const adviceManagerInstance = new AdviceManager();
