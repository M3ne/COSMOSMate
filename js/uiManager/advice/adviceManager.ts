/* /////////////////////////////////////////////////////////////////////////////

(c) 2016-2020, Theodore Kruczek

advice-module.js manages all recommended actions to the user in a semi-tutorial
manner. It works closely with ui.js
http://keeptrack.space

All rights reserved. No part of this web site may be reproduced, published,
distributed, displayed, performed, copied or stored for public or private
use, without written permission of the author.

No part of this code may be modified or changed or exploited in any way used
for derivative works, or offered for sale, or used to construct any kind of database
or mirrored at any other location without the express written permission of the author.

///////////////////////////////////////////////////////////////////////////// */

import { getEl } from '../../lib/helpers';
import { keepTrackApi } from '../../api/keepTrackApi';

let isAdviceEnabled = true;
let isAdviceOpen = false;
let helpOuterDOM: HTMLElement;
let helpHeaderDOM: HTMLElement;
let helpTextDOM: HTMLElement;
let tutIconDOM: HTMLElement;

export const isEnabled = (): boolean => isAdviceEnabled;
export const on = () => {
  try {
    localStorage.setItem('isAdviceEnabled', 'true');
  } catch {
    // Do Nothing
  }
  isAdviceEnabled = true;
  tutIconDOM.classList.add('bmenu-item-selected');
};
export const off = () => {
  try {
    localStorage.setItem('isAdviceEnabled', 'false');
  } catch {
    // Do Nothing
  }
  isAdviceEnabled = false;
  isAdviceOpen = false;
  helpOuterDOM.style.display = 'none';
  tutIconDOM.classList.remove('bmenu-item-selected');
};
export const showAdvice = (header: string, text: string) => {
  if (!isAdviceEnabled) return;

  isAdviceOpen = true;

  adviceManager.clearAdvice();

  helpOuterDOM.style.display = 'block';
  helpHeaderDOM.innerHTML = header;
  helpTextDOM.innerHTML = text;
  helpOuterDOM.addEventListener('click', function () {
    isAdviceOpen = false;
    helpOuterDOM.style.display = 'none';
  });
};
export const clearAdvice = function (): void {
  helpHeaderDOM.classList.remove('help-header-sel');
  helpHeaderDOM.onclick = null;
};
export const init = () => {
  helpOuterDOM = getEl('help-outer-container');
  helpHeaderDOM = getEl('help-header');
  helpTextDOM = getEl('help-text');
  tutIconDOM = getEl('tutorial-icon');

  tutIconDOM.addEventListener('click', function () {
    keepTrackApi.methods.onHelpMenuClick();
  });

  window.onkeydown = (e: KeyboardEvent) => {
    // If Shift + F1
    if (e.shiftKey && e.code === 'F1') {
      if (isAdviceOpen) {
        isAdviceOpen = false;
        helpOuterDOM.style.display = 'none';
      } else {
        keepTrackApi.methods.onHelpMenuClick();
      }
    }
  };
};

export const adviceManager = {
  clearAdvice: clearAdvice,
  isEnabled: isEnabled,
  off: off,
  on: on,
  showAdvice: showAdvice,
};
