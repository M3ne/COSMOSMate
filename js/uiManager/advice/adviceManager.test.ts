import * as adviceManager from './adviceManager';

// @ponicode
describe('adviceManager.isEnabled', () => {
  test('0', () => {
    adviceManager.init();
    adviceManager.on();
    let result: any = adviceManager.isEnabled();
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    adviceManager.init();
    adviceManager.off();
    let result: any = adviceManager.isEnabled();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('adviceManager.on', () => {
  test('0', () => {
    adviceManager.init();
    let result: any = adviceManager.on();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('adviceManager.off', () => {
  test('0', () => {
    adviceManager.init();
    let result: any = adviceManager.off();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('adviceManager.clearAdvice', () => {
  test('0', () => {
    adviceManager.init();
    let result: any = adviceManager.clearAdvice();
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('adviceManager.showAdvice', () => {
  beforeAll(() => {
    adviceManager.init();
    adviceManager.on();
  });
  test('0', () => {
    let result: any = adviceManager.showAdvice('test', 'test');
    expect(result).toMatchSnapshot();
  });
});
