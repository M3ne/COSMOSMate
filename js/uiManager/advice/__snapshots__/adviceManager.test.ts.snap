// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adviceManager.clearAdvice 0 1`] = `undefined`;

exports[`adviceManager.isEnabled 0 1`] = `true`;

exports[`adviceManager.isEnabled 1 1`] = `false`;

exports[`adviceManager.off 0 1`] = `undefined`;

exports[`adviceManager.on 0 1`] = `undefined`;

exports[`adviceManager.showAdvice 0 1`] = `undefined`;
